rspace/src/test/scala/coop/rchain/rspace/SerializeTests.scala

```diff
@@ -0,0 +1,39 @@
+package coop.rchain.rspace
+
+import coop.rchain.rspace.examples.StringExamples.Pattern
+import coop.rchain.rspace.examples.StringExamples.implicits._
+import coop.rchain.rspace.test.ArbitraryInstances._
+import coop.rchain.rspace.test.roundTripCodec
+import org.scalacheck.Prop
+import org.scalactic.anyvals.PosInt
+import org.scalatest.prop.{Checkers, Configuration}
+import org.scalatest.{FlatSpec, Matchers}
+import scodec.DecodeResult
+
+class SerializeTests extends FlatSpec with Matchers with Checkers with Configuration {
+
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    PropertyCheckConfiguration(minSuccessful = PosInt(1000))
+
+  "A String" should "round trip through a derived codec" in {
+
+    val propRoundTripCodec: Prop = Prop.forAll { (str: String) =>
+      roundTripCodec(str)(stringSerialize.toCodec)
+        .map((value: DecodeResult[String]) => value.value == str)
+        .getOrElse(default = false)
+    }
+
+    check(propRoundTripCodec)
+  }
+
+  "A Pattern" should "round trip through a derived codec" in {
+
+    val propRoundTripCodec: Prop = Prop.forAll { (pat: Pattern) =>
+      roundTripCodec(pat)(patternSerialize.toCodec)
+        .map((value: DecodeResult[Pattern]) => value.value == pat)
+        .getOrElse(default = false)
```

Comment:
I think you can factor that test out now. I've seen this in few places already. Minor thing but can become handy.